api/filter/accesslog.proto

```diff
@@ -53,120 +82,132 @@ message AccessLogEntry {
     // Request was ratelimited locally by rate limit filter.
     RATE_LIMITED = 11;
   }
-  // Status flags about the response.
-  repeated ResponseFlag response_flags = 4;
+  repeated ResponseFlag response_flags = 12;
 
-  // The upstream host URL (Envoy connects to).
-  //
-  // For example, tcp://ip:port for TCP connections.
+  // All metadata encountered during request processing, including endpoint
+  // selection.
   //
-  // IPv6 addresses should be stored in canonical (compressed) format using
-  // [address]:port notation.
-  string upstream_host = 5;
+  // This can be used to associate IDs attached to the various configurations
+  // used to process this request with the access log entry.  For example, a
+  // route created from a higher level forwarding rule with some ID can place
+  // that ID in this field and cross reference later. It can also be used to
+  // determine if a canary endpoint was used or not.
+  google.protobuf.Struct metadata = 13;
+}
 
-  // The Upstream Cluster that the upstream host belongs to.
-  string upstream_cluster = 6;
+message TLSProperties {
+  // TLS Version
+  enum TLSVersion {
+    VERSION_UNSPECIFIED = 0;
+    TLSv1 = 1;
+    TLSv1_1 = 2;
+    TLSv1_2 = 3;
+    TLSv1_3 = 4;
+  }
+  TLSVersion tls_version = 1;
 
-  // This field is the IP and port on which the request from the user was
-  // received, stored in ipv4:port or [ipv6]:port format.
-  string destination_host = 7;
+  // TLS Cipher suite negotiated during TLS handshake.
+  // The value is four hex digits defined by the IANA TLS Cipher Suite Registry,
+  // eg, "009C" for TLS_RSA_WITH_AES_128_GCM_SHA256.
+  //
+  // Here it is expressed as an integer.
+  google.protobuf.UInt32Value tls_cipher_suite = 2;
 
-  // Size of the HTTP request body in bytes
-  google.protobuf.UInt64Value request_body_bytes = 8;
+  // SNI hostname from handshake.
+  string tls_sni_hostname = 3;
+}
 
-  // Size of the HTTP response body in bytes
-  google.protobuf.UInt64Value response_body_bytes = 9;
+message TCPAccessLogEntry {
+  // Common properties shared by all Envoy access logs.
+  AccessLogBaseEntry base_entry = 1;
+}
 
-  // Size of the HTTP request headers in bytes
-  google.protobuf.UInt64Value request_headers_bytes = 10;
+message HTTPRequestProperties {
```

Comment:
While this is a breaking change at the wire level, I think this is OK as we haven't frozen the access log bits yet.